### PR TITLE
Added basic shapes example

### PR DESCRIPTION
**Pull request by chitalian**
This will hopefully show some people how to create shapes and anti-aliased lines

### CODE REVIEW

File: src/basic_shapes.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+//! An example of generating basic shapes
+extern crate image;
+extern crate num_complex;
+extern crate line_drawing;
+
+fn circle_bres(imgbuf: image::RgbaImage, xc: i32, yc: i32, r: i32) -> image::RgbaImage {
```
Comment (expenses) [inline src/basic_shapes.rs]: Change this to take `imgbuf` as `&mut`: `circle_bres(imgbuf: &mut image::RgbaImage, xc: i32, yc: i32, r: i32)`

File: src/basic_shapes.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	let mut imgbuf_clone = imgbuf.clone();
+	for (x, y) in line_drawing::BresenhamCircle::new(xc, yc, r) {
+		imgbuf_clone.put_pixel(x as u32, y as u32, image::Rgba([255, 255, 0, 255]));
+    }
+	return imgbuf_clone;
+}
+
+fn draw_lines(imgbuf: image::RgbaImage, lines: Vec<((u32, u32), (u32, u32))>, anti_alias: bool) -> image::RgbaImage {
```
Comment (expenses) [inline src/basic_shapes.rs]: Same here, and take `lines` as a slice: `&[((u32, u32), (u32, u32))]`.

File: src/basic_shapes.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	}
+	return imgbuf_clone;
+}
+
+fn triangle(imgbuf: image::RgbaImage, points: [(u32, u32); 3], anti_alias: bool) -> image::RgbaImage {
+	let lines = vec![
+		(points[0], points[1]),
+		(points[0], points[2]),
+		(points[1], points[2])
+	];
```
Comment (expenses) [inline src/basic_shapes.rs]: After you've done the above, you can change this to an array.

File: src/basic_shapes.rs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	// triangle
+	let tri_points = [
+		(400,100),
+		(600,300),
+		(200,300)
+	];
+	imgbuf = triangle(imgbuf, tri_points, true);
+
+	imgbuf = rectangle(imgbuf, (200, 300), (600, 700));
+	imgbuf = circle_bres(imgbuf, 80, 80, 50);
```
Comment (expenses) [inline src/basic_shapes.rs]: After the above, this should look something like
```rust
draw_triangle(&mut imgbuf, tri_points, true);
...
```

**Review by expenses — CHANGES_REQUESTED** [review]
Looks pretty good, I've made some suggestions though.

**Comment by chitalian** [comment]
@expenses Thanks a bunch for your feedback! I am still trying to learn the best Rust coding practices :) I made those changes, and want to ask if this file should live in /src or /examples

I originally intended for it to be in /examples but accidentally put it in /src (oops!)

**Comment by expenses** [comment]
No problem :^) yeah, it should go in `examples`.

**Comment by expenses** [comment]
Great, thanks!

**Comment by chitalian** [comment]
Thank you!

**Pull request merged**